File: examples/psamples.java
// file: psamples.java		G. Moody	11 March 2006 	
//
// Java translation of psamples.c from the WFDB Programmer's Guide

import wfdb.*;

public class psamples {

    public static void main(String argv[]) {
        WFDB_SiginfoArray siarray = new WFDB_SiginfoArray(2);
        if (wfdb.isigopen ("100s", siarray.cast(), 2) < 2)
            System.exit(1);
        WFDB_SampleArray v = new WFDB_SampleArray(2);
        for (int i = 0; i < 10; i++) {
            if (wfdb.getvec(v.cast()) < 0)
                break;
            System.out.println("\t" + v.getitem(0) + "\t" + v.getitem(1));
        }
    }
}

File: examples/psamples.pl
#!/usr/bin/perl
#
# file: psamples.pl		G. Moody	11 March 2006 	
#
# Perl translation of psamples.c from the WFDB Programmer's Guide 	

package wfdb;
use wfdb;

$siarray = new wfdb::WFDB_SiginfoArray(2);
if ($nsig = isigopen("100s", $siarray->cast(), 2) < 2) {
    exit(1);
}
$v = new wfdb::WFDB_SampleArray(2);
for ($i=0; $i < 10; $i++) {
    if (getvec($v->cast()) < 0) {
        exit(2);
    }
    print "\t", $v->getitem(0), "\t", $v->getitem(1), "\n";
}
